### PR TITLE
Convert C4 AI To Actual Minimax

### PR DESCRIPTION
**Pull request by TylerMathis**
C4 wasn't using real minimax before, and actually had a pretty significant bug in it's main evaluation module. This has been addressed, and now Lion is tough to beat! (but possible, remember that C4 is a solved game for white)

### CODE REVIEW

File: src/app/plugins/connectfour.plugin.ts [hunk 1]
```diff
@@ -204,38 +202,36 @@ class ConnectFourGame {
...
   private _lionMove() {
+    const bestCol = this._getBestMove();
+    this._dropPiece(bestCol);
+  }
+
+  // Turn on the beast.
+  private _getBestMove(): number {
     const moves: { col: number; val: number }[] = [];
     for (let col = 0; col < this._cols; col++) {
       if (!this._dropPiece(col)) {
```
Comment (tanndlin) [inline src/app/plugins/connectfour.plugin.ts:214]: does `this._dropPiece` affect the state of the board?

Comment (tanndlin) [reply src/app/plugins/connectfour.plugin.ts:214]: oh, i see the `this._removeTopPiece(col);`
You should prob comment that this change gets reversed at the bottom of the function

Comment (TylerMathis) [reply src/app/plugins/connectfour.plugin.ts:214]: Yeah I can notate that this is a backtracking algorithm.

**Review by suneettipirneni — APPROVED** [review]
lgtm

**Review by tanndlin — APPROVED** [review]

**Pull request merged**